sycl/source/detail/pi.cpp

```diff
@@ -37,22 +38,55 @@ std::string platformInfoToString(pi_platform_info info) {
 // Check for manually selected BE at run-time.
 bool useBackend(Backend TheBackend) {
   static const char *GetEnv = std::getenv("SYCL_BE");
-  static const Backend Use =
-    std::map<std::string, Backend>{
-      { "PI_OPENCL", SYCL_BE_PI_OPENCL },
-      { "PI_OTHER",  SYCL_BE_PI_OTHER }
+  static const Backend Use = std::map<std::string, Backend>{
+      {"PI_OPENCL", SYCL_BE_PI_OPENCL}, {"PI_OTHER", SYCL_BE_PI_OTHER}
       // Any other value would yield PI_OPENCL (current default)
-    }[ GetEnv ? GetEnv : "PI_OPENCL"];
+  }[GetEnv ? GetEnv : "PI_OPENCL"];
```

Comment:
What a hack instead of using another `?:` for example :-)

Reply:
Yes. :) 
It was already there. Took me an extra minute to understand what it was doing.

Reply:
Well, is it an obfuscation contest? :-)
Anyway I doubt this code can remain like this...
Some black hats can inject some radioactive string in SYCL_BE and then it inserts `Backend {}` in the `std::map` and return this value... What is a `Backend {}`? :-)
https://en.cppreference.com/w/cpp/container/map/operator_at
This selection logic should be part of a real Backend class, such as a static factory, that should throw or something like that.

Reply:
Thank you for noticing this.
the default constructor on enum Backend will have 0 in it, which corresponds to SYCL_BE_PI_OPENCL. So it will not be an issue.
The comment above the statement says: " // Any other value would yield PI_OPENCL (current default)"
Let me know if you feel this is still unsafe/obscure and would like me to change it.

Reply:
I hope you are kidding... :-)
What about launching a SYCL kernel to figure out what is the back-end to use, just in case a `std::map` memory allocation + insertion time & memory + obfuscation were not enough and too fast? :-)

Reply:
I understand that it is an overkill for a simple logic. I have edited the code. Let me know if the logic doesn't look better to you.
done.